### PR TITLE
Improve GeoJSON and image source handling

### LINKED ISSUE #145: add icon in map
I try to add an image to the map at the specified coordinates using addImage, but the error "Image has to contain an ID." keeps appearing. How can I solve this? I've put the code below.

image
image
image

### PR DESCRIPTION
**Pull request by ntadej**
- Make sure GeoJSON is properly parsed. This is a bit hacky as it has to go `QJSValue` -> `QVariant` -> `QJsonDocument` -> `QString` and then JSON is again parsed by MapLibre, but this works. It can be improved in the future if needed.
- Add `ImageParameter` to be able to add an image from Qt resources system (or local file) directly.

Implicitly closes #145 as I can not reproduce the issue.

### CODE REVIEW

File: src/location/plugins/declarative_style_parameter.hpp [hunk 1]
```diff
@@ -10,7 +10,12 @@
 #include <QtCore/QMetaObject>
 #include <QtCore/QMetaProperty>
 
-#define MLN_DECLARATIVE_PARSER(Class)                                                                           \
+#define MLN_DECLARATIVE_PARSER(Class) \
```
Comment (louwers) [inline src/location/plugins/declarative_style_parameter.hpp:13]: I don't believe this macro is worth it.

Comment (ntadej) [reply src/location/plugins/declarative_style_parameter.hpp:13]: In most cases there is no custom init method needed. Is there a possibility for an "ifdef" so I only call the second parameter if available?

Comment (louwers) [reply src/location/plugins/declarative_style_parameter.hpp:13]: Can't you use inheritance for componentComplete?

Comment (ntadej) [reply src/location/plugins/declarative_style_parameter.hpp:13]: Let me try again, but I remember that I was not possible (i.e. `QQmlParserStatus` needs to be direct base class).

Comment (ntadej) [reply src/location/plugins/declarative_style_parameter.hpp]: OK, I found a cleaner solution -> wrap the property access and convert if needed.

**Review by louwers — APPROVED** [review]
